Source/WebCore/platform/graphics/BitmapImageSource.cpp

```diff
@@ -609,7 +604,15 @@ Expected<Ref<NativeImage>, DecodingStatus> BitmapImageSource::nativeImageAtIndex
 Expected<Ref<NativeImage>, DecodingStatus> BitmapImageSource::currentNativeImageForDrawing(SubsamplingLevel subsamplingLevel, const DecodingOptions& options)
 {
     startAnimation(subsamplingLevel, options);
-    return nativeImageAtIndexForDrawing(currentFrameIndex(), subsamplingLevel, options);
+
+    auto effectiveOptions = options;
+
+    // If frame0 is displayed for the first time, startAnimation() has to request decoding frame1
+    // asynchronously. A flicker will occur if we request decoding frame0 also asynchronously.
+    if (UNLIKELY(options.decodingMode() == DecodingMode::Asynchronous && isAnimated() && !hasEverAnimated()))
```

Comment:
I don't think the `UNLIKELY` will do anything useful.

Reply:
`UNLIKELY` was removed.